src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/admin/AdminReclusterService.kt

```diff
@@ -0,0 +1,85 @@
+package uk.gov.justice.digital.hmpps.personrecord.api.controller.admin
+
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.retry.annotation.Backoff
+import org.springframework.retry.annotation.Retryable
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClientException
+import uk.gov.justice.digital.hmpps.personrecord.api.model.admin.AdminReclusterRecord
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.repository.PersonRepository
+import uk.gov.justice.digital.hmpps.personrecord.model.types.SourceSystemType
+import uk.gov.justice.digital.hmpps.personrecord.service.cprdomainevents.events.telemetry.RecordClusterTelemetry
+import uk.gov.justice.digital.hmpps.personrecord.service.message.recluster.TransactionalReclusterService
+import uk.gov.justice.digital.hmpps.personrecord.service.search.PersonMatchService
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
+
+@Component
+class AdminReclusterService(
+  private val transactionalReclusterService: TransactionalReclusterService,
+  private val personRepository: PersonRepository,
+  private val personMatchService: PersonMatchService,
+  private val publisher: ApplicationEventPublisher,
+) {
+
+  @Retryable(
+    maxAttempts = 2,
+    backoff = Backoff(delay = 200, random = true, multiplier = 3.0),
+    retryFor = [
+      WebClientException::class,
+    ],
+  )
```

Comment:
not sure this is the right place to retry - it will retry everything from the start I think